### ISSUE TITLE
Input Validation, conditional to specify if validity icon should be shown...

### ISSUE THREAD

**Issue by Lewiscowles1986** [fyne-io/fyne#2710]
### Is your feature request related to a problem? Please describe:

So I've been hacking at the Sudoku 😁 

```
var inputNumericOrBlankValidator = func(input string) error {
	if input != "1" && input != "2" && input != "3" && input != "4" && input != "5" && input != "6" && input != "7" && input != "8" && input != "9" && input != "" {
		return fmt.Errorf("invalid input")
	}
	return nil
}
```

This allows me to modify something like 

```
func (g *gui) newCell(x, y int) *widget.Entry {
	entry := &widget.Entry{}
	g.cells[x][y] = entry
	entry.Validator = inputNumericOrBlankValidator
	return entry
}
```

One of the issues I've had is that this outputs an annoying tick on every input.

![image](https://user-images.githubusercontent.com/2605791/147655259-055af2e7-9f53-452f-b460-55b42065b691.png)

### Is it possible to construct a solution with the existing API?

Not AFAIK `fyne\v2@v2.1.2\widget\entry_validation.go` contains the following

```
        if r.entry.validationError == nil && r.entry.Text != "" {
		r.icon.Resource = theme.ConfirmIcon()
		r.icon.Show()
	} else if r.entry.validationError != nil && !r.entry.focused && r.entry.dirty {
		r.icon.Resource = theme.NewErrorThemedResource(theme.ErrorIcon())
		r.icon.Show()
	} else {
		r.icon.Hide()
	}
```

I Believe I would like to supply an option so that theme.ConfirmIcon() is not set to be the resource unless `ShowConfirmation` (open to better name) is set to true; which would need to cascade from every input type down to the icon that is part of entry validation 😭 .

### Describe the solution you'd like to see:

Ironically you already have it with the guard for `r.entry.Text != ""`. I Just want another conditional so that the tick is not shown if I use a validator; only the error icon, in cases of malformed input.

**Comment by Lewiscowles1986** [comment]
I have a patch "hacked" case to fix.

```
func (g *gui) newCell(x, y int) *widget.Entry {
	entry := &widget.Entry{}
	entry.Validator = inputNumericOrBlankValidator
	entry.OnlyShowValidationErrors = true
	g.cells[x][y] = entry
	return entry
}
```

This uses edits to `fyne.io\fyne\v2@v2.1.2\widget\entry.go` (only the `OnlyShowValidationErrors` entry on Struct)

```
	Validator                fyne.StringValidator `json:"-"`
	validationStatus         *validationStatus
	onValidationChanged      func(error)
	validationError          error
	OnlyShowValidationErrors bool
```

Then an edit to the validator `fyne.io\fyne\v2@v2.1.2\widget\entry_validation.go`

`if r.entry.validationError == nil && r.entry.Text != "" && !r.entry.OnlyShowValidationErrors {`

**Comment by Lewiscowles1986** [comment]
I think this is a two-liner if acceptable 😉

**Comment by andydotxyz** [comment]
Hmm, this is interesting. However if you want to hide the icon of validation would it not make more sense to hide both of them (OK and Failure)? 
Otherwise it is reserving a very big space that may never get used if the user does not make a mistake.

**Comment by andydotxyz** [comment]
I wonder if this relates to a possible accessory API that validation and the password dialog both use - that might be more flexible, though not certain it matches our API design of being semantic based.

**Comment by Lewiscowles1986** [comment]
I'm literally kicking the tyres as I try out various experiments with each of the repo's. Testing the fences; seeing what language and concepts are available.

As you mentioned, I should use a custom widget for this. Also in the docs, I did notice a rather nice numeric input; which I believe restricts the valid inputs responded to, and more accurately than validation prevents invalid characters, without iconography. [NumericalEntry](https://developer.fyne.io/extend/numerical-entry)

As you do note, there is an empty icon sitting doing nothing in success cases, but it was already there for empty cases, and is used for error cases. It seems to be set from a validator being present, rather than from state being valid / invalid (it's a tri-state with blank omitting an icon as the outcome seems indeterminate). This does conflict with the validator you'll see I wrote; which includes an empty string input.

The reason I quite liked showing only error cases was that it was quite unlikely someone would be able to continue to operate with incorrect entry values; but the information density was also high for sudokgo on a 9x9 grid; that the success icons distracted. If the grid were to support larger (hexidecimal) grids, the problem would be worse; and NumericalEntry would likely need to be further specialized. It was likely a dead-end. I'll close this out.
